### PR TITLE
mint with comment

### PR DESCRIPTION
**Pull request by jgeary**

### CODE REVIEW

File: src/ERC721Drop.sol [hunk 1]
```diff
@@ -597,7 +640,8 @@ contract ERC721Drop is
             to: _msgSender(),
             quantity: quantity,
             pricePerToken: pricePerToken,
-            firstPurchasedTokenId: firstMintedTokenId
+            firstPurchasedTokenId: firstMintedTokenId,
+            comment: comment
```
Comment (oveddan) [inline src/ERC721Drop.sol:644]: good call not storing this and just emitting it.

**Review by oveddan — COMMENTED** [review]
awesome feature!

File: src/ERC721Drop.sol [hunk 2]
```diff
@@ -564,6 +580,33 @@ contract ERC721Drop is
...
+    {
+        return _handlePurchasePresale(quantity, maxQuantity, pricePerToken, merkleProof, comment);
+    }
+
+    function _handlePurchasePresale(
+        uint256 quantity,
+        uint256 maxQuantity,
+        uint256 pricePerToken,
+        bytes32[] calldata merkleProof,
+        string memory comment
```
[the remark this answers is not in the document]
Comment (oveddan) [reply src/ERC721Drop.sol:608]: got it - ok

[the remark this answers is not in the document]
Comment (jgeary) [reply src/ERC721Drop.sol:608]: oh disregard about the test file i got confused bc was also addressing your comment on the test file

File: test/ERC721Drop.t.sol [hunk 3]
```diff
@@ -300,6 +308,14 @@ contract ERC721DropTest is Test {
         uint256 paymentAmount = uint256(salePrice) * purchaseQuantity + zoraFee;
         vm.deal(address(456), paymentAmount);
         vm.prank(address(456));
+        vm.expectEmit(true, true, true, true);
+        emit Sale(
+            address(456),
+            purchaseQuantity,
+            salePrice,
+            0,
+            ""
```
[the remark this answers is not in the document]
Comment (jgeary) [reply test/ERC721Drop.t.sol:317]: sg

File: src/ERC721Drop.sol [hunk 4]
```diff
@@ -597,7 +640,8 @@ contract ERC721Drop is
             to: _msgSender(),
             quantity: quantity,
             pricePerToken: pricePerToken,
-            firstPurchasedTokenId: firstMintedTokenId
+            firstPurchasedTokenId: firstMintedTokenId,
```
Comment (iainnash) [inline src/ERC721Drop.sol:643]: I would strongly caution against changing this event.

File: yarn.lock [hunk 5]
```diff
@@ -1247,14 +1247,6 @@ is-wsl@^2.1.1:
   dependencies:
     is-docker "^2.0.0"
 
-isomorphic-fetch@^3.0.0:
```
[the remark this answers is not in the document]
Comment (jgeary) [reply yarn.lock:1250]: running `yarn`

File: foundry.toml [hunk 6]
```diff
@@ -2,6 +2,7 @@
 solc_version = '0.8.17'
 optimizer = true
 optimizer_runs = 5000
+via_ir = true
```
Comment (iainnash) [inline foundry.toml:5]: :cries_in_compile_time:

File: broadcast/DeployNewERC721Drop.s.sol/5/run-1681831282.json
Comment (iainnash) [inline broadcast/DeployNewERC721Drop.s.sol/5/run-1681831282.json:1]: can we strike these files from the PR?

**Review by iainnash — APPROVED** [review]
looks good! some small comments on style.

**Pull request merged**